x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/TDigestParser.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.analytics.mapper;
+
+import org.elasticsearch.index.mapper.DocumentParsingException;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+public class TDigestParser {
+
+    private static final ParseField COUNTS_FIELD = new ParseField("counts");
+    private static final ParseField VALUES_FIELD = new ParseField("centroids");
```

Comment:
If you are going for a different member-name on the former `values` field, you should probably also rename the constant:

```suggestion
    private static final ParseField CENTROIDS_FIELD = new ParseField("centroids");
```